src/Geometry/PUMLReader.h

```diff
@@ -94,6 +88,81 @@ class PUMLReader : public MeshReader
 	static int FIRST_FACE_VERTEX[4];
 };
 
+/*
+ * Possible types of boundary conditions for SeisSol.
+ */
+enum class BCType {
+  internal, external, unknown
+};
+
+/*
+ * Decodes the boundary condition tag into a BCType.
+ */
+inline BCType bcToType(int id) {
+  if (id == 0 || id == 3 || id > 64) {
+    return BCType::internal;
+  } else if (id == 1 || id == 2 || id == 5 || id == 6) {
+    return BCType::external;
+  } else {
+    return BCType::unknown;
+  }
+}
+
+/*
+ * Decodes the boundary condition tag into a string representation.
+ */
+inline std::string bcToString(int id) {
+  if (id == 0) { return std::string("regular"); }
+  else if (id == 1) { return std::string("free surface"); }
+  else if (id == 2) { return std::string("free surface with gravity"); }
+  else if (id == 3) { return std::string("dynamic rupture"); }
+  else if (id == 5) { return std::string("absorbing"); }
+  else if (id == 6) { return std::string("periodic"); }
+  else if (id > 64) {
+    std::stringstream s;
+    s << "fault-tagging (" << id << ")";
+    return s.str();
+  } else {return std::string(""); }
+}
+
+/*
+ * Check if the mesh is locally correct:
+ * - if a face is an internal face, the neighbor has to exist.
+ * - if a face is an external face, the neighbor must not exist.
+ * @param face: face to check
+ * @param cellNeighbors: ids of the neighboring cells
+ * @param side: side of the tetrahedron to check
+ * @param sideBC: boundary condition tag at the side to check
+ * @param cellIdAsInFile: Original cell id as it is given in the h5 file
+ */
+inline bool checkMeshCorrectnessLocally(PUML::TETPUML::face_t face, int* cellNeighbors, int side, int sideBC, int cellIdAsInFile) {
```

Comment:
checkBoundaryConditionsOfCell?

Reply:
Hmm, I don't really agree, IMHO `checkBoundaryConditionsOfCell` could also imply that we only check that some BC is set. I think Correctness is better here, since it implies that there are some correct and some incorrect meshes.

Reply:
ok.
my point is that this function only checks one face of the mesh (but I guess this is what is meant by Locally).
Let's keep it as it is then.